elixir/apps/domain/test/domain/policies_test.exs

```diff
@@ -0,0 +1,371 @@
+      assert {:error, changeset} = create_policy(%{}, subject)
+
+      assert changeset.errors == [
+               name: {"can't be blank", [validation: :required]},
+               actor_group_id: {"can't be blank", [validation: :required]},
+               resource_id: {"can't be blank", [validation: :required]}
+             ]
+    end
+
+    test "returns changeset error on invalid params", %{subject: subject} do
+      assert {:error, changeset} =
+               create_policy(
+                 %{name: 1, actor_group_id: "foo", resource_id: "bar"},
+                 subject
+               )
+
+      assert changeset.errors == [
+               name: {"is invalid", [{:type, :string}, {:validation, :cast}]}
+             ]
+    end
+
+    test "returns error when subject has no permission to manage policies", %{subject: subject} do
+      subject = AuthFixtures.remove_permissions(subject)
+
+      assert create_policy(%{}, subject) ==
+               {:error,
+                {:unauthorized,
+                 [
+                   missing_permissions: [
+                     {:one_of, [Policies.Authorizer.manage_policies_permission()]}
+                   ]
+                 ]}}
+    end
+
+    test "returns changeset error when trying to create policy with another account actor_group",
+         %{
+           account: account,
+           subject: subject
+         } do
+      other_account = AccountsFixtures.create_account()
+
+      resource = ResourcesFixtures.create_resource(account: account)
+      other_actor_group = ActorsFixtures.create_group(account: other_account)
+
+      assert {:error, changeset} =
+               create_policy(
+                 %{
+                   account_id: account.id,
+                   name: "yikes",
+                   actor_group_id: other_actor_group.id,
+                   resource_id: resource.id
+                 },
+                 subject
+               )
+
+      assert changeset.errors == [
+               actor_group:
+                 {"does not exist",
+                  [
+                    {:constraint, :assoc},
+                    {:constraint_name, "policies_actor_group_id_fkey"}
+                  ]}
+             ]
+    end
+
+    test "returns changeset error when trying to create policy with another account resource", %{
+      account: account,
+      subject: subject
+    } do
+      other_account = AccountsFixtures.create_account()
+
+      other_resource = ResourcesFixtures.create_resource(account: other_account)
+      actor_group = ActorsFixtures.create_group(account: account)
+
+      assert {:error, changeset} =
+               create_policy(
+                 %{
+                   account_id: account.id,
+                   name: "yikes",
+                   actor_group_id: actor_group.id,
+                   resource_id: other_resource.id
+                 },
+                 subject
+               )
+
+      assert changeset.errors == [
```

Comment:
There is a helper `errors_on(changeset)` that we should use not to compare opaque values within `changeset.errors`. Eg.:
```
assert errors_on(changeset) == [ .. ]
```

Reply:
Gotcha.  Will get all of this updated to use `errors_on`

Reply:
This should be updated